### PR TITLE
feat(packages): upgrade to Emotion v10, Downshift to v4, Jest to v24

### PR DESCRIPTION
**Pull request by rachmulvey**
This PR upgrades Emotion to next major version v10. This upgrade required the following changes:

- Upgrade Jest and Downshift dependencies
- Use `getRootProps` method from Downshift
- Update Jest setup
- Remove shallow rendering in tests and replace with mounts and component mocks
- Minor object style changes
- Use <Global/> from emotion rather than updating DOM elements with emotion styles directly

---
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/178)

### CODE REVIEW

**Comment by tysonwolker** [comment]
Just check bundle size, I realise our bundling sucks but just curious the impact.

File: setupTests.js [hunk 1]
```diff
@@ -1,10 +1,6 @@
 /* eslint-disable no-unused-vars */
 import { configure } from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
-import { createSerializer } from 'jest-emotion';
-import * as emotion from 'emotion';
-
-expect.addSnapshotSerializer(createSerializer(emotion));
```
Comment (congcongcong250) [inline setupTests.js]: noice

File: src/components/InfoSection/__tests__/InfoSection.spec.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+
 describe('InfoSection', () => {
   it('renders correctly with props', () => {
     const requiredProps = {
       content: 'mock content',
       height: '100px',
       width: '200px'
     };
-    const component = shallow(<InfoSection {...requiredProps} />);
+    const component = mount(<InfoSection {...requiredProps} />);
```
Comment (congcongcong250) [inline src/components/InfoSection/__tests__/InfoSection.spec.js]: why do we have to mount may i ask?

File: package.json [hunk 3]
```diff
@@ -36,10 +36,10 @@
   },
   "homepage": "https://github.com/qantasairways/runway#readme",
   "dependencies": {
+    "@emotion/core": "^10.0.27",
```
Comment (congcongcong250) [inline package.json]: do we have to declare dependency if it's a peerdependency?

File: src/components/InfoSection/__tests__/InfoSection.spec.js (hunk 2 above)
Comment (rachmulvey) [reply src/components/InfoSection/__tests__/InfoSection.spec.js]: shallow isn't supported by emotion 10

File: package.json (hunk 3 above)
Comment (rachmulvey) [reply package.json]: good catch, removed!

**Comment by rachmulvey** [comment]
> Just check bundle size, I realise our bundling sucks but just curious the impact.

It adds 18kb ungzipped, 6 is from the Downshift upgrade and 12 from emotion upgrade

File: package.json (hunk 3 above)
Comment (rachmulvey) [reply package.json]: Good catch, I've moved the peer deps from dependencies to devDependencies because we still need them to get installed in CI

**Pull request merged**